### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by VitaliiLeshchenko**

### CODE REVIEW

File: src/main/java/threads/ThreadExtendedThread.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    private static final Logger LOGGER = Logger.getLogger(ThreadExtendedThread.class);
+    private final Counter counter;
+
+    public ThreadExtendedThread(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+        while (counter.getCounter() < 100) {
```
Comment (RomanDubovskyi) [inline src/main/java/threads/ThreadExtendedThread.java:15]: 100 magic number, let's make it constant

File: src/main/java/threads/ThreadImplementsRunnable.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+public class ThreadImplementsRunnable implements Runnable {
+    private static final Logger LOGGER = Logger.getLogger(ThreadImplementsRunnable.class);
+    private final Counter counter;
+
+    public ThreadImplementsRunnable(Counter counter) {
+        this.counter = counter;
+    }
+
+    public void run() {
+        while (counter.getCounter() < 100) {
```
Comment (RomanDubovskyi) [inline src/main/java/threads/ThreadImplementsRunnable.java:14]: same here

Comment (VitaliiLeshchenko) [reply src/main/java/threads/ThreadImplementsRunnable.java]: done

File: src/main/java/threads/ThreadExtendedThread.java (hunk 1 above)
Comment (VitaliiLeshchenko) [reply src/main/java/threads/ThreadExtendedThread.java]: done

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                    <encoding>UTF-8</encoding>
+                    <consoleOutput>true</consoleOutput>
+                    <failsOnError>true</failsOnError>
+                    <linkXRef>false</linkXRef>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>
```
[the remark this answers is not in the document]
Comment (VitaliiLeshchenko) [reply pom.xml]: done

**Pull request merged**